samples/MegaDrone/src/main/cpp/MegaDroneEngine.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <memory>
+#include "MegaDroneEngine.h"
+
+MegaDroneEngine::MegaDroneEngine(std::vector<int> cpuIds) : AudioEngine(std::make_shared<DefaultAudioStreamCallback>()) {
+    mDefaultCallbackPtr->setParent(*this);
```

Comment:
This relationship between `MegaDroneEngine` and `AudioEngine`, along with the child and parent callback objects is difficult to understand. It took me several minutes to figure it out and I'm familiar with the code. We need to make this simpler to avoid creating a barrier for new Oboe users. One idea is to have  `MegaDroneEngine` be composed of an `AudioEngine` rather than being one. Let's discuss.

Reply:
I created a branch with this change. You can compare it with your current branch here: https://github.com/google/oboe/compare/refactor-samples...refactor-samples-don?expand=1